### PR TITLE
PP-5170 Process charge capture from sqs message

### PR DESCRIPTION
**Pull request by sfount**
* call capture charge on capture queue messages 
* models messages with Jackson library

Co-authored-by: @jankowiakmaria

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/queue/CaptureQueue.java [hunk 1]
```diff
@@ -39,11 +49,31 @@ public void sendForCapture(String externalId) throws QueueException {
...
+            try {
+                CaptureCharge captureCharge = jsonObjectMapper
+                        .getObject(qm.getMessageBody(), CaptureCharge.class);
+                return ChargeCaptureMessage.of(captureCharge, qm);
+            } catch (WebApplicationException e) {
+                logger.warn("Error parsing the charge capture message from queue [{}]", e.getMessage());
+                return null;
+            }
+        };
+    }
```
Comment (mrlumbu) [inline src/main/java/uk/gov/pay/connector/queue/CaptureQueue.java:69]:
```
public List<ChargeCaptureMessage> retrieveChargesForCapture() throws QueueException {
        List<QueueMessage> queueMessages = sqsQueueService
                .receiveMessages(this.captureQueueUrl, CAPTURE_MESSAGE_ATTRIBUTE_NAME);

        return queueMessages
                .stream()
                .map(this::toChargeCaptureMessage)
                .filter(Objects::nonNull)
                .collect(Collectors.toList());
    }

    private ChargeCaptureMessage toChargeCaptureMessage(QueueMessage qm) {
        try {
            CaptureCharge captureCharge = jsonObjectMapper
                    .getObject(qm.getMessageBody(), CaptureCharge.class);
            return ChargeCaptureMessage.of(captureCharge, qm);
        } catch (WebApplicationException e) {
            logger.warn("Error parsing the charge capture message from queue [{}]", e.getMessage());
            return null;
        }
    }
```
is cleaner

File: src/main/java/uk/gov/pay/connector/queue/CaptureQueue.java [hunk 2]
```diff
@@ -39,11 +49,31 @@ public void sendForCapture(String externalId) throws QueueException {
...
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
     }
-    
-    public void markMessageAsProcessed(QueueMessage message) throws QueueException {    
+
+    private Function<QueueMessage, ChargeCaptureMessage> getChargeCaptureMessageFunction() {
+        return qm -> {
+            try {
+                CaptureCharge captureCharge = jsonObjectMapper
```
Comment (mrlumbu) [inline src/main/java/uk/gov/pay/connector/queue/CaptureQueue.java]: it occurs to me that this mapping from string to object would more nicely done as a static factory in CaptureCharge class, but that's a minor thing

File: src/main/java/uk/gov/pay/connector/queue/sqs/ChargeCaptureMessage.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package uk.gov.pay.connector.queue.sqs;
```
Comment (mrlumbu) [inline src/main/java/uk/gov/pay/connector/queue/sqs/ChargeCaptureMessage.java]: don't think this is sqs specific. I'm quite keen to try and separate interface and implementation for this queue stuff. Finessing that can wait, but we should at least aim to keep implementation agnostic classes out of the implementation package

**Review by mrlumbu — CHANGES_REQUESTED** [review]
Looks good. Few minor points

File: src/main/java/uk/gov/pay/connector/queue/CaptureQueue.java (hunk 1 above)
Comment (sfount) [reply src/main/java/uk/gov/pay/connector/queue/CaptureQueue.java:69]: Agreed! Great suggestion

File: src/main/java/uk/gov/pay/connector/queue/CaptureQueue.java (hunk 2 above)
Comment (sfount) [reply src/main/java/uk/gov/pay/connector/queue/CaptureQueue.java]: Did consider this - can be improved in a further PR I think

**Review by mrlumbu — APPROVED** [review]

**Pull request merged**